### PR TITLE
Added tests for commons utilities and fixed FormatUtils bug.

### PR DESCRIPTION
**Pull request by freddytuxworth**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?
there will be

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
it will do

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?
it will have been

- [ ] Is your initial contribution a single, squashed commit?
it is

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?

I analyzed this project and noticed some methods lacking tests in these two utilities classes. I've written some tests using [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/datasheet) which should hopefully help you to detect any regressions caused by future code changes. 

I also corrected a small bug in `FormatUtils` which caused the `formatNanos` method to incorrectly format multiples of 1000000 or 1000000000 to `"0 nanos"`.

Please let me know if there are any other areas of your project which you would like tests written for, I would be more than happy to help.

### CODE REVIEW

File: nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/FormatUtils.java [hunk 1]
```diff
@@ -399,8 +399,8 @@ private static String join(final String delimiter, final String... values) {
     public static String formatNanos(final long nanos, final boolean includeTotalNanos) {
         final StringBuilder sb = new StringBuilder();
 
-        final long seconds = nanos > 1000000000L ? nanos / 1000000000L : 0L;
-        long millis = nanos > 1000000L ? nanos / 1000000L : 0L;
+        final long seconds = nanos >= 1000000000L ? nanos / 1000000000L : 0L;
+        long millis = nanos >= 1000000L ? nanos / 1000000L : 0L;
```
Comment (peterschrammel) [inline nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/FormatUtils.java:403]: Fix in separate commit

File: nifi-commons/nifi-utils/src/test/java/org/apache/nifi/processor/TestFormatUtils.java [hunk 2]
```diff
@@ -57,4 +57,42 @@ public void testFormatTime() throws Exception {
...
+        assertEquals("0 nanos (0 nanos)", FormatUtils.formatNanos(0L, true));
+
+        assertEquals("1 millis, 0 nanos", FormatUtils.formatNanos(1_000_000L, false));
+        assertEquals("1 millis, 0 nanos (1000000 nanos)", FormatUtils.formatNanos(1_000_000L, true));
+
+        assertEquals("1 millis, 1 nanos", FormatUtils.formatNanos(1_000_001L, false));
+        assertEquals("1 millis, 1 nanos (1000001 nanos)", FormatUtils.formatNanos(1_000_001L, true));
+
+        assertEquals("1 seconds, 0 millis, 0 nanos", FormatUtils.formatNanos(1_000_000_000L, false));
+        assertEquals("1 seconds, 0 millis, 0 nanos (1000000000 nanos)", FormatUtils.formatNanos(1_000_000_000L, true));
```
Comment (peterschrammel) [inline nifi-commons/nifi-utils/src/test/java/org/apache/nifi/processor/TestFormatUtils.java]: break after 80 chars

File: nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/FormatUtils.java (hunk 1 above)
Comment (freddytuxworth) [reply nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/FormatUtils.java:403]: This is now in a separate commit. The PR text does ask for "a single, squashed commit" though, but I suppose they are quite separate issues.

File: nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StringUtilsTest {
+
+  @Test
+  public void testIsBlank() {
+    Assert.assertEquals(false, StringUtils.isBlank("0"));
+    Assert.assertEquals(true, StringUtils.isBlank(" "));
+    Assert.assertEquals(true, StringUtils.isBlank(""));
```
Comment (sudo-jaa) [inline nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java]: I'd be interested to see an assertion here using `\u0000` as an input, possibly both "\u0000", and " \u0000 ". I would also quite like to see an assertion for a non-printable character that is not one of the following recognised java whitespace characters: 

"
     ('\u00A0', '\u2007', '\u202F').
     '\t', U+0009 HORIZONTAL TABULATION.
     '\n', U+000A LINE FEED.
     '\u000B', U+000B VERTICAL TABULATION.
     '\f', U+000C FORM FEED.
     '\r', U+000D CARRIAGE RETURN.
     '\u001C', U+001C FILE SEPARATOR.
     '\u001D', U+001D GROUP SEPARATOR.
     '\u001E', U+001E RECORD SEPARATOR.
     '\u001F', U+001F UNIT SEPARATOR. "

File: nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package org.apache.nifi.util;
+
+import java.util.ArrayList;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StringUtilsTest {
+
+  @Test
+  public void testIsBlank() {
```
Comment (sudo-jaa) [inline nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java:11]: Change these Assertions to `assertTrue` / `assertFalse`

File: nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    Assert.assertEquals(false, StringUtils.isBlank("\u0000"));
+    Assert.assertEquals(false, StringUtils.isBlank(" \u0000 "));
+    Assert.assertEquals(false, StringUtils.isBlank(" \u5678 "));
+    
+    Assert.assertEquals(true, StringUtils.isBlank(" "));
+    Assert.assertEquals(true, StringUtils.isBlank(""));
+  }
+
+  @Test
+  public void testStartsWith() {
```
Comment (sudo-jaa) [inline nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java:22]: Change these Assertions to `assertTrue`

File: nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  public void testPadLeft() {
+    Assert.assertEquals("sample", StringUtils.padLeft("sample", 0, '0'));
+    Assert.assertEquals("0000sample", StringUtils.padLeft("sample", 10, '0'));
+    Assert.assertEquals("0000000000", StringUtils.padLeft("", 10, '0'));
+
+    Assert.assertNull(StringUtils.padLeft(null, 0, '0'));
+  }
+
+  @Test
+  public void testIsEmpty() {
```
Comment (sudo-jaa) [inline nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java:49]: Change these Assertions to `assertTrue` (or `assertFalse`)

File: nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    Assert.assertEquals(false, StringUtils.isBlank(" \u0000 "));
+    Assert.assertEquals(false, StringUtils.isBlank(" \u5678 "));
+    
+    Assert.assertEquals(true, StringUtils.isBlank(" "));
+    Assert.assertEquals(true, StringUtils.isBlank(""));
+  }
+
+  @Test
+  public void testStartsWith() {
+    Assert.assertEquals(false, StringUtils.startsWith("!", "something"));
```
Comment (sudo-jaa) [inline nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java]: Can we add an `Assert.assertFalse(StringUtils.startsWith("!!!!!test", "!!!!"))`? 

(I think that returns false, I haven't checked properly. Should be an interesting case either way)

File: nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+    Assert.assertNull(StringUtils.padLeft(null, 0, '0'));
+  }
+
+  @Test
+  public void testIsEmpty() {
+    Assert.assertEquals(false, StringUtils.isEmpty("AAAAAAAA"));
+
+    Assert.assertEquals(true, StringUtils.isEmpty(""));
+    Assert.assertEquals(true, StringUtils.isEmpty(null));
```
Comment (sudo-jaa) [inline nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java]: Add a test for a single whitespace input.

File: nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  @Test
+  public void testSubstringAfter() {
```
Comment (sudo-jaa) [inline nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java:58]: Should we add an assertion for what happens when a string has multiple of the same separators?

File: nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    final ArrayList<String> collection = new ArrayList<>();
+    assertEquals("", StringUtils.join(collection, ","));
+
+    collection.add("test1");
+    assertEquals("test1", StringUtils.join(collection, ","));
+
+    collection.add("test2");
+    assertEquals("test1,test2", StringUtils.join(collection, ","));
+
+    collection.add(null);
```
Comment (sudo-jaa) [inline nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/StringUtilsTest.java:78]: Perhaps add an assertion for when we add a new item with a different delimiter?
